mongodbatlas/resource_mongodbatlas_federated_settings_identity_provider.go

```diff
@@ -0,0 +1,198 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/spf13/cast"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceMongoDBAtlasFederatedSettingsIdentityProvider() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderRead,
+		ReadContext:   resourceMongoDBAtlasFederatedSettingsIdentityProviderRead,
+		UpdateContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderUpdate,
+		DeleteContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasFederatedSettingsIdentityProviderImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"federation_settings_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"associated_domains": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"sso_debug_enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"okta_idp_id": {
```

Comment:
Again this is missing quite a few options - ssoUrl, requestBinding, etc... https://www.mongodb.com/docs/atlas/reference/api/identity-provider-update-one/
And shouldn't okta_idp_id just be idp_id?

Reply:
shorten to idp_id

Reply:
added missing parameters